pkg/verifier/gcp/entry_point.go

```diff
@@ -23,40 +29,77 @@ const (
 // - find unreachable endpoints & parse output, then terminate instance
 // - return `g.output` which stores the execution results
 func (g *GcpVerifier) ValidateEgress(vei verifier.ValidateEgressInput) *output.Output {
+	// Default to curl.Probe if no Probe specified
+	if vei.Probe == nil {
+		vei.Probe = curl.Probe{}
+		g.Logger.Debug(vei.Ctx, "defaulted to curl probe")
+	}
 	g.Logger.Debug(vei.Ctx, "Using configured timeout of %s for each egress request", vei.Timeout.String())
 	//default gcp machine e2
 	if vei.InstanceType == "" {
 		vei.InstanceType = DEFAULT_INSTANCE_TYPE
 	}
+	if vei.Timeout == 0 {
+		vei.Timeout = DEFAULT_TIMEOUT
+	}
 
+	vei.Timeout = DEFAULT_TIMEOUT
```

Comment:
This section looks a little confusing - it looks like `vei.Timeout` will always get set to `DEFAULT_TIMEOUT` no matter what. What's the intention here for setting that timeout?

Reply:
Changed to `if vei.Timeout <= 0` and only setting timeout there based on those conditions.